From bd60226a17b974225be7b5a34f4145a94c5053a9 Mon Sep 17 00:00:00 2001
From: Ziyuan Xu <xzy.xu@rock-chips.com>
Date: Wed, 4 Nov 2020 17:21:31 +0800
Subject: [PATCH] ARM: dts: rv1126-bat-ipc-v10: tune suitable memory for
 isp/ispp

For SC210IOT (1080p normal raw10) sensor:
isp_reserved size = 5 * dmatx_buffer + 5 * isp_ispp_buffer + 3*1024*1024
dmatx_buffer = ALIGN(width * bpp / 8, 256) * height = 2560 *1080
isp_ispp_buffer =
    ALIGN(width, 16) * ALIGN(height, 16) * (1/16 + ((is_yuv422) ? 2 : 1.5)) +
    ALIGN(width, 64) * ALIGN(height, 128) / 16
    = 1920*1088 * 1.5625 + 1920 * 1088 * 0.0625

isp_reserved(CMA) size = 32.3 MB
CMA buffer should be aligned as 4MB, isp_reserved size should be 36MB
for SC210IOT.

Signed-off-by: Ziyuan Xu <xzy.xu@rock-chips.com>
Change-Id: I839c7ffd179f35f6d6b9b581eda0837e631420bd
---
 arch/arm/boot/dts/rv1126-bat-ipc-v10.dts | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/rv1126-bat-ipc-v10.dts b/arch/arm/boot/dts/rv1126-bat-ipc-v10.dts
index 7ff3f9faa8e7..76e5b24f34b4 100644
--- a/arch/arm/boot/dts/rv1126-bat-ipc-v10.dts
+++ b/arch/arm/boot/dts/rv1126-bat-ipc-v10.dts
@@ -273,7 +273,7 @@
 };
 
 &isp_reserved {
-	size = <0x05c00000>;
+	size = <0x02400000>;
 };
 
 &npu {
@@ -338,7 +338,7 @@
 };
 
 &rkisp_thunderboot {
-	reg = <0x10000000 (72 * 0x00100000)>;
+	reg = <0x10000000 (36 * 0x00100000)>;
 };
 
 &rkisp_vir0 {
-- 
2.35.3

